src/mono/wasm/runtime/binding_support.js

```diff
@@ -666,6 +732,394 @@ var BindingSupportLib = {
+				if (conv.needs_root)
+					needs_root_buffer = true;
+				localStep.needs_root = conv.needs_root;
+				localStep.key = args_marshal[i];
+				steps.push (localStep);
+				size += conv.size;
+			}
+
+			return { 
+				steps: steps, size: size, args_marshal: args_marshal, 
+				is_result_definitely_unmarshaled: is_result_definitely_unmarshaled,
+				is_result_possibly_unmarshaled: is_result_possibly_unmarshaled,
+				result_unmarshaled_if_argc: result_unmarshaled_if_argc,
+				needs_root_buffer: needs_root_buffer
+			};
+		},
+
+		_get_converter_for_marshal_string: function (args_marshal) {
+			// console.log("_get_converter_for_marshal_string", args_marshal);
+
+			if (!this._signature_converters)
+				this._signature_converters = new Map();
+
+			var converter = this._signature_converters.get (args_marshal);
+			if (!converter) {
+				converter = this._create_converter_for_marshal_string (args_marshal);
+				this._signature_converters.set (args_marshal, converter);
+			}
+
+			// console.log("result.args_marshal", converter.args_marshal);
+
+			return converter;
+		},
+
+		_compile_converter_for_marshal_string: function (args_marshal) {
+			// console.log("_compile_converter_for_marshal_string", args_marshal);
+
+			var converter = this._get_converter_for_marshal_string (args_marshal);
+			if (typeof (converter.args_marshal) !== "string")
+				throw new Error ("Corrupt converter for '" + args_marshal + "'");
+
+			if (converter.compiled_function && converter.compiled_variadic_function)
+				return converter;
+
+			var converterName = args_marshal.replace("!", "_result_unmarshaled");
+			converter.name = converterName;
+			
+			var body = [];
+			var argumentNames = ["buffer", "rootBuffer", "method"];
+
+			// worst-case allocation size instead of allocating dynamically, plus padding
+			var bufferSizeBytes = converter.size + (args_marshal.length * 4) + 16;
+			var rootBufferSize = args_marshal.length;
+			// ensure the indirect values are 8-byte aligned so that aligned loads and stores will work
+			var indirectBaseOffset = ((((args_marshal.length * 4) + 7) / 8) | 0) * 8;
+
+			var closure = {};
+			var indirectLocalOffset = 0;
+
+			body.push (
+				"if (!buffer) buffer = Module._malloc (" + bufferSizeBytes + ");",
+				"var indirectStart = buffer + " + indirectBaseOffset + ";",
+				"var indirect32 = (indirectStart / 4) | 0, indirect64 = (indirectStart / 8) | 0;",
+				"var buffer32 = (buffer / 4) | 0;",
+				""
+			);
+
+			for (let i = 0; i < converter.steps.length; i++) {
+				var step = converter.steps[i];
+				var closureKey = "step" + i;
+				var valueKey = "value" + i;
+
+				var argKey = "arg" + i;
+				argumentNames.push (argKey);
+
+				if (step.convert) {
+					closure[closureKey] = step.convert;
+					// body.push ("console.log('calling converter " + step.key + "', " + closureKey + ", 'with value', " + argKey + ");"); 
+					body.push ("var " + valueKey + " = " + closureKey + "(" + argKey + ", method, " + i + ");");
```

Comment:
String interpolation makes this a lot easier to read, I think it worth doing

```suggestion
					body.push (`var ${valueKey} = ${closureKey} (${argKey} , method, ${i});`);
```

Reply:
Unfortunately it does need a scheme/host, it used to work without it but now unless it's a well-formed URL the chrome dev tools get confused